Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs

```diff
@@ -82,12 +96,190 @@ sendPaymentReminderToDriver Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId
+  firstJobOfWindow <- getDriverFeeCalcJobFlagKey startTime endTime merchantId
+  when (firstJobOfWindow == Just True) $ do
+    driverFees <- findFeesInRangeWithStatus startTime endTime ONGOING Nothing
+    for_ driverFees $ \driverFee -> do
+      mbDriverPlan <- findByDriverId (cast driverFee.driverId)
+      plan <- getPlan mbDriverPlan merchantId
+      cacheDriverPlan driverFee.driverId endTime plan
+      Hedis.del (mkDriverFeeCalcJobFlagKey startTime endTime merchantId)
+  -- Schedule notif job
+
+  now <- getCurrentTime
+  transporterConfig <- SCT.findByMerchantId merchantId >>= fromMaybeM (TransporterConfigNotFound merchantId.getId)
+  driverFees <- findFeesInRangeWithStatus startTime endTime ONGOING transporterConfig.driverFeeCalculatorBatchSize
+
+  for_ driverFees $ \driverFee -> do
+    mbPlan <- getDriverPlanCache endTime driverFee.driverId
+    plan <- case mbPlan of
+      Nothing -> throwError $ InternalError ("No plan found for driver" <> driverFee.driverId.getId)
+      Just plan_ -> pure plan_
+
+    let (planBaseFrequcency, baseAmount) = getFreqAndBaseAmountcase plan.planBaseAmount
+    let due = fromIntegral driverFee.govtCharges + fromIntegral driverFee.platformFee.fee + driverFee.platformFee.cgst + driverFee.platformFee.sgst
+    driver <- QP.findById (cast driverFee.driverId) >>= fromMaybeM (PersonDoesNotExist driverFee.driverId.getId)
+
+    (totalFee, offerId) <- case planBaseFrequcency of
+      "PER_RIDE" -> do
+        let numRides = driverFee.numRides - plan.freeRideCount
+            feeWithoutDiscount = min plan.maxAmount (baseAmount * HighPrecMoney (toRational numRides))
+        getFinalOrderAmount feeWithoutDiscount merchantId transporterConfig driver plan.id baseAmount driverFee.id
+      "DAILY" -> do
+        let numRides = driverFee.numRides - plan.freeRideCount
+            feeWithoutDiscount = if numRides > 0 then baseAmount else 0
+        getFinalOrderAmount feeWithoutDiscount merchantId transporterConfig driver plan.id baseAmount driverFee.id
+      _ -> return (0, Nothing) -- TODO: handle WEEKLY and MONTHLY later
+    updateOfferId offerId driverFee.id now
+    unless (totalFee == 0) $ do
+      driverFeeSplitter plan totalFee transporterConfig driverFee now
+      updatePendingPayment True (cast driverFee.driverId)
+
+    -- blocking
+    dueInvoices <- QDF.findAllPendingAndDueDriverFeeByDriverId (cast driverFee.driverId) -- Problem with lazy evaluation?
+    let driverFeeIds = map (.id) dueInvoices
+    when (due + totalFee >= plan.maxCreditLimit) $ do
+      updateStatus PAYMENT_OVERDUE now driverFee.id
+      updateFeeType RECURRING_INVOICE now `mapM_` driverFeeIds
+      updateSubscription False (cast driverFee.driverId)
+      QDFS.updateStatus (cast driverFee.driverId) DDFS.PAYMENT_OVERDUE -- only updating when blocked. Is this being used?
+  case listToMaybe driverFees of
+    Nothing -> return Complete
+    _ -> case transporterConfig.driverFeeCalculatorBatchGap of
```

Comment:
why have it as Maybe if Nothing means error ? if backward compatibility is the concern then its not an issue for a such a small table, we can just add the required value before release.

Reply:
It is only used for a specific merchant, no other merchant will trigger this scheduler